Fog.js

```diff
@@ -2358,7 +2354,7 @@ function bucketFill(ctx, mouseX, mouseY, fogStyle = 'rgba(0,0,0,0)', fogType=0){
 	let distance = 10000;
   	particleUpdate(mouseX, mouseY); // moves particle
 	particleLook(ctx, window.walls, distance, fog, fogStyle, fogType); 
-	redraw_light_walls();
+	redraw_light_walls(ctx);

```

Comment:
This errors since it's reading ctx as the canvas variable in redraw_light_walls.

Maybe we just pass on ctx to redraw_light_walls? If ctx is undefined then grab the canvas and context? I think anywhere we can pass on the canvas we also have the context of temp_overlay.

Reply:
Ill play a little with this and see which looks better